arquidiocesis-back/routes/estadisticas.js

```diff
@@ -36,18 +32,113 @@ const getEstadisticas = async (firestore, req, res) => {
     profesional: 0,
   };
 
-  var total = 0;
+  let total = 0;
+
+  const miembrosRef = firestore.collection('miembros');
+  let miembros = null;
+
+  const zone = req.query.zone || null;
+  // Filter by zone
+  if (zone) {
+    const zonesRef = firestore.collection('zonas');
+    const zoneDoc = await zonesRef.doc(zone).get();
+
+    const decanatoRef = firestore.collection('decanatos');
+    const decanato = req.query.decanato || null;
+    const decanatoIds = [];
+    // Filter by decanato
+    if (decanato) {
+      decanatoIds.push(decanato);
+    } else {
+      const decanatoDocs = await decanatoRef
+        .where('zona', '==', zoneDoc.id)
+        .get();
+      decanatoDocs.forEach((doc) => decanatoIds.push(doc.id));
+    }
+
+    // Get Parroquias
+    const parroquiasRef = firestore.collection('parroquias');
+
+    let batches = [];
+    while (decanatoIds.length) {
+      // Firestore limits batches to 10
+      const ids = decanatoIds.splice(0, 10);
+
+      // Add request to queue
+      batches.push(
+        new Promise((response) => {
+          parroquiasRef
+            .where('decanato', 'in', ids)
+            .get()
+            .then((parroquiaDocs) => {
+              const resultIds = [];
+              parroquiaDocs.forEach((doc) => resultIds.push(doc.id));
+              response(resultIds);
+            });
+        })
+      );
+    }
+
+    // After all of the data is fetched, return it
+    const parroquiaIds = await (await Promise.all(batches)).flat();
+
+    // Get Grupos
+    const gruposRef = firestore.collection('grupos');
+
+    batches = [];
+    while (parroquiaIds.length) {
+      // Firestore limits batches to 10
+      const ids = parroquiaIds.splice(0, 10);
+
+      // Add request to queue
+      batches.push(
+        new Promise((response) => {
+          gruposRef
+            .where('parroquia', 'in', ids)
+            .get()
+            .then((grupoDocs) => {
+              const resultIds = [];
+              grupoDocs.forEach((doc) => resultIds.push(doc.id));
+              response(resultIds);
+            });
+        })
+      );
+    }
+    const grupoIds = await (await Promise.all(batches)).flat();
+
+    // Get miembros
+    batches = [];
+    while (grupoIds.length) {
+      // Firestore limits batches to 10
+      const ids = grupoIds.splice(0, 10);
+
+      // Add request to queue
+      batches.push(
+        new Promise((response) => {
+          miembrosRef
+            .where('grupo', 'in', ids)
+            .get()
+            .then((miembrosDocs) => {
+              response(miembrosDocs.docs);
+            });
+        })
+      );
+    }
+
+    miembros = await (await Promise.all(batches)).flat();
```

Comment:
Creo que estan de más los awaits fuera de la promesa, los métodos de los arreglos no son asíncronos 😄 
```suggestion
    miembros = (await Promise.all(batches)).flat();
```